components/ArticleForm.js

```diff
@@ -0,0 +1,213 @@
+      id: id || '',
+      link: link || '',
+      portals,
+    };
+
+    await articleCallback(formData);
+
+    router.push('/');
+  };
+
+  const keywordPropSeparated = useMemo(
+    () => keywords?.join?.(', ') || '',
+    [keywords],
+  );
+
+  const transformedPortals = useMemo(
+    () => listOfPortalsValues.map((item) => !!portals?.includes(item)),
+    [portals],
+  );
+
+  return (
+    <form className="mt-5" onSubmit={formSubmit}>
+      <div className="flex flex-col justify-between lg:flex-row">
+        <div className="w-full lg:w-[70%]">
+          <FormInput
+            labelClassName="my-1 block text-sm text-stone-700"
+            label="Název"
+            className="headline--2 mb-6 block w-full border-b-2 border-stone-300 bg-light-50 p-2 text-stone-700"
+            id="title"
+            type="text"
+            required
+            defaultValue={title || ''}
+            pattern="[a-zA-Z0-9 .]+"
+            title="Pouze velka a mala pismena a cisla."
+          />
+
+          <FormTextarea
+            labelClassName="mt-6 mb-1 block text-sm text-stone-700"
+            className="mb-4 block w-full border-b-2 border-stone-300 bg-light-50 p-2"
+            label="Obsah"
+            cols="60"
+            rows="20"
+            id="content"
+            required
+            defaultValue={content || ''}
+          />
+
+          <FormTextarea
+            labelClassName="my-1 block text-sm text-stone-700"
+            id="keywords"
+            className="mb-6 border-b-2 border-stone-300 bg-light-50 p-2"
+            label=" Klíčová slova - více slov oddělte čárkou"
+            rows="4"
+            cols="30"
+            defaultValue={keywordPropSeparated || ''}
+            pattern="[a-zA-Z0-9 ,]+"
+            title="Pouze velka a mala pismena nebo cisla."
+          />
+
+          <div>
+            <label
+              className="my-1 mt-6 block text-sm text-stone-700"
+              htmlFor="portals"
+            >
+              Portály
+            </label>
+            <ul>
+              {listOfPortalsValues.map((item, index) => (
+                <li key={index}>
+                  <FormCheckbox
+                    className="mr-3 mb-2"
+                    type="checkbox"
+                    id={`portals[${index}]`}
+                    value={item}
+                    // checked={portalsCheckboxes[index]}
+                    // onChange={() => checkboxesOnChangeHandler(index)}
+                    defaultChecked={transformedPortals[index]}
+                  />
+                  <label htmlFor={`portals-${index}`}>{item}</label>
+                </li>
+              ))}
+            </ul>
+          </div>
+        </div>
+
+        <aside className="w-full lg:w-[25%]">
+          <FormCheckbox
+            id="isTitleDifferent"
+            checked={isTitleDifferent}
+            onChange={handleTitleCheckboxChange}
+            defaultChecked={isTitleDifferent || false}
```

Comment:
This does not work the way you imagine. 

Since you are passing inside `checked` which is NOT undefined. The logic inside the `FormCheckbox` -> 

``` 
const [isChecked, setChecked] = useState(defaultChecked);

checked={checked === undefined ? isChecked : checked}
```

completely ignores the defaultValue. 

You either pass in `value` + `onChange` or none + `defaultChecked`. Not both. Otherwise you would get this - >

`
App contains an input of type text with both value and defaultValue props. Input elements must be either controlled or uncontrolled (specify either the value prop, or the defaultValue prop, but not both). Decide between using a controlled or uncontrolled input element and remove one of these props. More info: https://reactjs.org/link/controlled-components
`